### ISSUE TITLE
Default values set incorrectly when passed context does not exist

### ISSUE THREAD

**Issue by KucharczykL** [wrabit/django-cotton#116]
# Steps to reproduce

1. Set a variable default inside a component using `<c-vars />`.
2. Include the component and pass the variable from the current context using `:variable=localvar`.
3. Inside the component, the variable is not considered empty for things like `{% if %}` AND the content of the variable is the name of the variable that should have contained it (for example `localvar` as in the example above).

# What do you expect to happen

I expect that component variables with defaults are not considered empty and their value is that of the default value when the passed context is undefined.

# Example

`index.html`
```django
<c-button :color=button.color />
<!-- button.color doesn't exist in this context -->
```

`cotton/button.html`
```django
<c-vars color="gray" />
{{ color }}
<!-- color is "button.color" -->
{% if color == "gray" %}
<!-- does not get executed -->
{% endif %}
```

**Comment by wrabit** [comment]
@KucharczykL Fixed in v0.9.36

```python
    def test_unprocessable_dynamic_attributes_fallback_to_cvars_defaults(self):
        self.create_template(
            "cotton/unprocessable_dynamic_attribute.html",
            """
                <c-vars color="gray" />
                {{ color }}
            """,
        )

        self.create_template(
            "unprocessable_dynamic_attribute_view.html",
            """
                <c-unprocessable-dynamic-attribute :color="button.color" />
            """,
            "view/",
            context={},
        )

        with self.settings(ROOT_URLCONF=self.get_url_conf()):
            response = self.client.get("/view/")
            self.assertTrue("gray" in response.content.decode())

**Comment by KucharczykL** [comment]
Thank you very much for such a fast response!
